### PR TITLE
Unify WebGL markers and rects

### PR DESCRIPTION
**Pull request by ianthomas23**
This PR unifies WebGL markers and rects as a `rect` is a `square` marker with a 2D size. It closed item 5 of #11369.

It simplifies the codebase as there is now just a single pair of shaders (`marker.vert` and `marker.frag`) for all constant-shape glyphs rather than there being separate ones for markers and rects. I have renamed the shaders `marker.frag` etc rather than `markers.frag` because the singular name is more appropriate as the equivalent typescript class is `MarkerGL`.

`rect` already had all visual properties except for line dashes. Now other markers have the same, particularly hatching and end caps, and I have rewritten many of the shader distance equations to fit in with the `rect` approach and also to make them more understandable/maintainable by using similar formulations of distances from straight lines defined by unit vectors and distances from the origin. This removes the last of the original WebGL code which had a different antialiasing formulation.

The only things missing from rects/markers now are:
   1. Line dashes, which needs to be correctly implemented for `line` first.
   2. Line joins (although `rect` already has them).

The WebGL baseline images remain the same for `rect` but are all different for other `marker`s due to the changes in antialiasing and the underlying equations. There is one new visual test, for markers with hatching:
Marker_glyph__should_support_hatch

This work is part of the CZI round 3 extension.

### CODE REVIEW

File: bokehjs/test/integration/glyphs/markers.ts [hunk 1]
```diff
@@ -112,4 +113,39 @@ describe("Marker glyph", () => {
...
+      "vertical_dash", "spiral", "right_diagonal_line", "left_diagonal_line", "diagonal_cross",
+      "right_diagonal_dash", "left_diagonal_dash", "horizontal_wave", "vertical_wave",
+      "criss_cross", "dot", "ring", "horizontal_line", "vertical_line", "cross",
+    ]
+
+    const markers: MarkerType[] = [
+      "circle", "circle_cross", "circle_dot", "circle_x", "circle_y", "diamond", "diamond_cross",
+      "diamond_dot", "hex", "hex_dot", "inverted_triangle", "square", "square_cross", "square_dot",
+      "square_pin", "square_x", "star", "star_dot", "triangle", "triangle_dot", "triangle_pin",
+    ]
```
Comment (mattpap) [inline bokehjs/test/integration/glyphs/markers.ts:129]: This should be imported from `core/enums` and then use e.g. `MarkerType.values`.

File: bokehjs/test/integration/glyphs/markers.ts [hunk 2]
```diff
@@ -112,4 +113,39 @@ describe("Marker glyph", () => {
...
+
+    function make_plot(output_backend: OutputBackend) {
+      const p = fig([300, 300],
+        {output_backend, title: output_backend, x_range: [-0.5, 4.5], y_range: [-0.6, 4.6]})
```
Comment (mattpap) [inline bokehjs/test/integration/glyphs/markers.ts:133]: To match the existing code style, this:
```ts
const p = fig([300, 300], {
  output_backend, title: output_backend,
  x_range: [-0.5, 4.5], y_range: [-0.6, 4.6],
})
```
or this
```ts
const p = fig([300, 300], {
  output_backend, title: output_backend,  x_range: [-0.5, 4.5], y_range: [-0.6, 4.6],
})
```
is preferred.

Comment (ianthomas23) [reply bokehjs/test/integration/glyphs/markers.ts:133]: Will do.

File: bokehjs/test/integration/glyphs/markers.ts (hunk 1 above)
Comment (ianthomas23) [reply bokehjs/test/integration/glyphs/markers.ts:129]: The test only uses a subset of all the marker types as some do not have any fill and therefore do not have any hatch (e.g. `asterisk`, `dash`, `x`, etc). If I used `MarkerType.values` I would have to filter them anyway, so I either have to name the ones I do want or the ones I don't want.

**Comment by ianthomas23** [comment]
This is ready, apart from @mattpap's comment about the marker types in the test which I am inclined to leave as it is.

File: bokehjs/test/integration/glyphs/markers.ts (hunk 1 above)
Comment (mattpap) [reply bokehjs/test/integration/glyphs/markers.ts:129]: I see. Well, we could provide enums of finer granularity in both cases, but that can be left for future PRs.

**Pull request merged**